pkg/storage/indexheader/stream_binary_reader_test.go

```diff
@@ -91,6 +91,7 @@ func TestStreamBinaryReader_CheckSparseHeadersCorrectnessExtensive(t *testing.T)
 
 				// Check correctness of sparse index headers.
 				compareIndexToHeader(t, b, r2)
+				compareIndexToHeaderPostings(t, b, r2)
```

Comment:
can you embed this into the `comapreIndexToHeader`? I see `TestReadersComparedToIndexHeader` is also suffering from the same